examples/resources/cloudflare_pages_project/resource.tf

```diff
@@ -98,3 +98,89 @@ resource "cloudflare_pages_project" "deployment_configs" {
     }
   }
 }
+
+# A Pages project managing all configs
+
+resource "cloudflare_pages_project" "deployment_configs" {
+  account_id        = var.account_id
+  name              = "this-is-my-project-01"
+  production_branch = "main"
+
+  source {
+    type = "github"
+    config {
+      owner                         = "cloudflare"
+      repo_name                     = "ninjakittens"
+      production_branch             = "main"
+      pr_comments_enabled           = true
+      deployments_enabled           = true
+      production_deployment_enabled = true
+      preview_deployment_setting    = "custom"
+      preview_branch_includes       = ["dev", "preview"]
+      preview_branch_excludes       = ["main", "prod"]
+    }
+  }
+
+  build_config {
+    build_command       = "npm run build"
+    destination_dir     = "build"
+    root_dir            = ""
+    web_analytics_tag   = "cee1c73f6e4743d0b5e6bb1a0bcaabcc"
+    web_analytics_token = "021e1057c18547eca7b79f2516f06o7x"
+  }
+
+  deployment_configs {
+    preview {
+      environment_variables = {
+        ENVIRONMENT = "preview"
+      }
+      secrets = {
+        TURNSTILE_SECRET = var.turnstile_secret
+      }
+      kv_namespaces = {
+        KV_BINDING = "5eb63bbbe01eeed093cb22bb8f5acdc3"
+      }
+      durable_object_namespaces = {
+        DO_BINDING = "5eb63bbbe01eeed093cb22bb8f5acdc3"
+      }
+      r2_buckets = {
+        R2_BINDING = "some-bucket"
+      }
+      d1_databases = {
+        D1_BINDING = "445e2955-951a-4358-a35b-a4d0c813f63"
+      }
+      compatibility_date  = "2022-08-15"
+      compatibility_flags = ["preview_flag"]
+    }
+    production {
+      environment_variables = {
+        ENVIRONMENT = "production"
+        OTHER_VALUE = "other value"
+      }
+      secrets = {
+        TURNSTILE_SECRET = var.turnstile_secret
```

Comment:
let's use fake but usable values here. assuming people are using variables makes this a little harder to grok for new comers.